axotag/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+#![deny(missing_docs)]
+#![allow(clippy::result_large_err)]
+
+//! # axotag
+//!
+//! This library contains tag-parsing code for use with cargo-dist.
+
+use errors::{TagError, TagResult};
+use semver::Version;
+
+pub mod errors;
+
+/// Represents an opaque package.
+pub struct Package {
+    /// The package's name
+    pub name: String,
+    /// The package's version, if specified
+    pub version: Option<Version>,
+}
+
+/// details on what we're announcing (partially computed)
+pub struct PartialAnnouncementTag {
+    /// The full tag
+    pub tag: String,
+    /// The release
+    pub release: ReleaseType,
+    /// whether we're prereleasing
+    pub prerelease: bool,
+}
+
+impl Default for PartialAnnouncementTag {
+    /// Constructs an empty PartialAnnouncementTag
+    fn default() -> PartialAnnouncementTag {
+        PartialAnnouncementTag {
+            tag: String::new(),
```

Comment:
yeah i had been thinking about that but "using String::new as secret Option::None" always seems like it causes me a problem down the road so I didn't want to lead you down the path of my crimes (fuck it, ship it, PartialAnnouncementTag is a temp type anyway).